File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.java
import java.util.HashMap;

class Solution {
    public String[] solution(String[] record) {
        int count = 0;
        HashMap<String, String> hashMap = new HashMap<>();
        
        for(String str : record){
            String[] pstr = str.split(" ");
            if (pstr[0].equals("Enter") || pstr[0].equals("Change"))
                hashMap.put(pstr[1], pstr[2]);
            if (pstr[0].equals("Change"))
                count++;
        }

        int i = 0;
        String[] result = new String[record.length - count];
        
        for(String str : record){
            String[] pstr = str.split(" ");
            if (pstr[0].equals("Enter"))
                result[i++] = hashMap.get(pstr[1]) + "님이 들어왔습니다.";
            else if (pstr[0].equals("Leave"))
                result[i++] = hashMap.get(pstr[1]) + "님이 나갔습니다.";
        }

        return result;
    }
}

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    answer = []
    dic = {}

    for s in record:
        s_split = s.split()
        if len(s_split) == 3:
            dic[s_split[1]] = s_split[2]
    
    for s in record:
        s_split = s.split()
        if s_split[0] == 'Enter':
            answer.append('%s님이 들어왔습니다.' %dic[s_split[1]])
        elif s_split[0] == 'Leave':
            answer.append('%s님이 나갔습니다.' %dic[s_split[1]])

    return answer